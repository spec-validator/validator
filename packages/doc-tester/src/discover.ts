import path from 'path'
import fs from 'fs'

export default (root: string, pattern: RegExp): string[] => {
  const result: string[] = []

  const _fromDir = (parent: string) =>
    fs.readdirSync(parent).forEach(file => {
      const filename = path.join(parent, file)
      if (fs.lstatSync(filename).isDirectory()) {
        _fromDir(filename)
      } else if (pattern.test(filename)) {
        result.push(filename)
      }
    })

  if (fs.existsSync(root)){
    _fromDir(root)
  }

  return result
}
