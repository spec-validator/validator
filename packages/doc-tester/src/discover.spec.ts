import discover from './discover'

test('with existing path', () => {
  expect(discover(`${__dirname}/fixtures`, /^.*\.md$/)).toMatchSnapshot()
})

test('with missing path', () => {
  expect(discover(`${__dirname}/missing`, /^.*\.md$/)).toMatchSnapshot()
})
