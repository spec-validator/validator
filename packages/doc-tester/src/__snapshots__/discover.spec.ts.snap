// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with existing path 1`] = `
Array [
  "/home/gurunars/repo/validator/packages/doc-tester/src/fixtures/README.md",
  "/home/gurunars/repo/validator/packages/doc-tester/src/fixtures/inner/ONE.md",
]
`;

exports[`with missing path 1`] = `Array []`;
